src/editor/decorations.ts

```diff
@@ -144,7 +150,12 @@ export function buildDecorations(
 		inCommentBlock: false,
 	};
 
-	for (const {from, to} of view.visibleRanges) {
+	for (let {from, to} of view.visibleRanges) {
+
+		// Extend parse range
+		from = Math.max(from - extraParseRange, 0);
+		to = Math.min(to + extraParseRange, view.state.doc.length);
```

Comment:
Also maybe this should be `view.state.doc.length - 1`? I'm not entirely sure how the from / to values work.